docs/samples/Microsoft.ML.Samples/Static/LightGBMMulticlassWithInMemoryData.cs

```diff
@@ -0,0 +1,163 @@
+﻿using Microsoft.ML.Data;
+using Microsoft.ML.LightGBM.StaticPipe;
+using Microsoft.ML.StaticPipe;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.ML.Samples.Static
+{
+    class LightGBMMulticlassWithInMemoryData
+    {
+        /// <summary>
+        /// Number of features per example used in <see cref="MultiClassLightGbmStaticPipelineWithInMemoryData"/>.
+        /// </summary>
+        private const int _featureVectorLength = 10;
+
+        /// <summary>
+        /// Data point used in <see cref="MultiClassLightGbmStaticPipelineWithInMemoryData"/>. A data set there 
+        /// is a collection of <see cref="NativeExample"/>.
+        /// </summary>
+        private class NativeExample
+        {
+            [VectorType(_featureVectorLength)]
+            public float[] Features;
+            [ColumnName("Label")]
+            public string Label;
+            public uint LabelIndex;
+            public uint PredictedLabelIndex;
+            [VectorType(4)]
+            // The probabilities of being "AA", "BB", "CC", and "DD".
+            public float[] Scores;
+
+            public NativeExample()
+            {
+                Features = new float[_featureVectorLength];
+            }
+        }
+
+        /// <summary>
+        /// Helper function used to generate <see cref="NativeExample"/>s.
+        /// </summary>
+        private static List<NativeExample> GenerateRandomExamples(int count)
```

Comment:
>   private static List<NativeExample> GenerateRandomExamples(int count) [](start = 5, length = 71)

I'd move this in the SampleUtils project, and reference it from the test, and from here.

Reply:
If we move this to SamplesUtils, then user will not able to see a full example in doc.

---
In reply to: [244875572](https://github.com/dotnet/machinelearning/pull/1953#discussion_r244875572) [](ancestors = 244875572)

Reply:
i believe the main purpose of those examples in the docs are to get copy/pasted in a VS project, add the necessary references and study it there, otherwise int he web page many things don't make sense. 

There is also guideline to keep those examples small, and too the point, otherwise their length is an immediate  turnoff for average user. 

---
In reply to: [245743767](https://github.com/dotnet/machinelearning/pull/1953#discussion_r245743767) [](ancestors = 245743767,244875572)

Reply:
Ok. Will move NativeExample and its generator to SamplesUtils.

---
In reply to: [245796925](https://github.com/dotnet/machinelearning/pull/1953#discussion_r245796925) [](ancestors = 245796925,245743767,244875572)